### ISSUE TITLE
Pydantic V2 JSON Schema lacks type attribute for `Literal`s

### ISSUE THREAD

**Issue by aradkdj** [pydantic/pydantic#6241]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2 installed directly from the `main` branch, or equivalent

### Description

Pydantic V1 inferred the types a `Literal` is compatible of and made them available in the JSON Schema. This is no longer the case with Pydantic V2.

### Example Code

```Python
import json
from typing import Literal

from pydantic import BaseModel


class Health(BaseModel):
    status: Literal["ok", "error"]

# With Pydantic V2
print(json.dumps(Health.model_json_schema(), indent=2))
#{
#  "properties": {
#    "status": {
#      "enum": [
#        "ok",
#        "error"
#      ],
#      "title": "Status"   
#    } <--- Notice properties.status has no type key
#  },
#  "required": [
#    "status"
#  ],
#  "title": "Health",
#  "type": "object"
#}

# When using V1
print(Health.schema_json(indent=2))
#{
#  "title": "Health",
#  "type": "object",
#  "properties": {
#    "status": {
#      "title": "Status",
#      "enum": [
#        "ok",
#        "error"
#      ],
#      "type": "string" <--- Notice properties.status.type is set to "string"
#    }
#  },
#  "required": [
#    "status"
#  ]
#}
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.0b3
        pydantic-core version: 0.39.0 release build profile
                 install path: /home/***/.local/share/virtualenvs/***/lib/python3.10/site-packages/pydantic
               python version: 3.10.11 (main, Apr 20 2023, 13:40:31) [GCC 13.0.1 20230401 (Red Hat 13.0.1-0)]
                     platform: Linux-6.3.8-200.fc38.x86_64-x86_64-with-glibc2.37
     optional deps. installed: ['email-validator', 'typing-extensions']
```

**Comment by adriangb** [comment]
The `type` key for `enum` is not required (see the examples at https://json-schema.org/). Is this causing any actual issues for you?

**Comment by aradkdj** [comment]
My primary use of Pydantic is through FastAPI. With the V1 behavior the OpenAPI schema was rendered as expected if I was to use Health as a path operation response model. A little more info in https://github.com/tiangolo/fastapi/discussions/9709#discussioncomment-6247094

**Comment by adriangb** [comment]
Gotcha. Shouldn’t be hard to add, but I am also curious if the new swagger update fixes this. Of course for something like `Literal[“foo”, 124]` we can’t add a type.

**Comment by adriangb** [comment]
Fixed in #6243
